internal/luks2/cryptsetup.go

```diff
@@ -255,6 +256,27 @@ func (options *FormatOptions) appendArguments(args []string) []string {
 	return args
 }
 
+var runtimeGOARCH = runtime.GOARCH
+
+// selectCipher will return the cipher to use. This is aes-xts-plain64
+// everywhere except on armhf (32bit arm) hardware where the XTS mode
+// cannot be accelerated by the hardware.
+//
+// Note that this is a simple approach, we could run "cryptsetup
+// benchmark" but that seems over engineered (and easy enough to
+// switch if we need in the future).
+func selectCipherAndKeysize() (string, int) {
+	switch runtimeGOARCH {
+	case "arm":
+		// on 32bit arm CPUs xts cannot be accerlated in HW so
+		// cbc is used
+		return "aes-cbc-essiv:sha256", keySize * 4
```

Comment:
Whilst this creates the correct key size, use of the keySize variable here is a bit weird because that's currently 64, which is the key size for AES-256-XTS in bytes (hence the `keySize * 8` that exists to convert it to bits). It might be better to get rid of the keySize variable and have a function that returns the correct value based on the selected cipher.

Reply:
Thank you, yeah, my approach is a bit awkward indeed, I reworked it now based on your suggestion :)